woocommerce/woocommerce-ios
# Add/Edit Products: main screen - add tappable Product description table view cell

Fixes #1420 

## Changes
- Created `PlaceholderOrTitleAndTextTableViewCell` as the cell that either shows a placeholder or a title+text UI (example design in #1420)
- Created `enum ProductFormSection` to describe each section on the Product form UI, and its own rows (`enum PrimaryFieldRow` and `enum DetailRow`
- Created protocol `ProductFormTableViewModel` for an array of sections for Product Form
  - Implemented `ProductFormTableViewModel` in `DefaultProductFormTableViewModel` given a Product
- Created `ProductFormTableViewDataSource` as the data source of the Product form table view. I'm not very sure if the use of `enum ProductFormSection` makes it easy to configure a table view UI, and any suggestions/thoughts are welcome!
- Coordinated the table view data source and table view model in `ProductFormViewController`
- Moved the timing of dismissing the `EditorViewController` to its `onContentSave` callback, and only dismissed the editor after the Product description is saved

## Testing
- Go to Products tab
- Tap a Product with non-empty description
- Tap "Edit" in the navigation bar --> there should be a row showing the current description in a single line
- Tap on the Product description row
- Remove all the description
- Tap "Done" --> after it goes back to the Edit Product screen, the description should be empty on the web

## Example screenshots

placeholder | non-empty description
-- | --
![Simulator Screen Shot - iPhone SE - 2019-11-14 at 11 44 09](https://user-images.githubusercontent.com/1945542/68825177-2dd52c00-06d4-11ea-922e-5ac0be27a1e3.png) | ![Simulator Screen Shot - iPhone SE - 2019-11-14 at 11 44 35](https://user-images.githubusercontent.com/1945542/68825178-2dd52c00-06d4-11ea-8660-f800ab6638bf.png)

Update release notes:
